### ISSUE TITLE
Improvements!  

### ISSUE THREAD

**Issue by ghost** [Apple0726/helixteus-3#37]
I can't use discord but I really liked the game and I wanted to share some ideas for improvements and balances: 

-  NOTE: I haven't "completed" the game 100% so some of these might be added but I don't know about it
- Nerf early game ship drives slightly
- Buff laser mining for rocks (wall mining speed is ok)
- Nerf damage for the player and enemies in ship combat and cave combat
- Nerf armor
- Make driving in caves smoother and faster on better wheels

- Remove cave hotbar (its kind of useless and drills and portals should be stored in the inventory with a button to access it) 
- Buff storage a bit
- Make machine mining wayyy faster 
- Allow for removing of obstacles like ice and hydrogen
- Add a way to mass select things on other guis like how you can mass turn on stone crushers

- I have physical pain when trying to dodge enemy attacks
- There is a glitch with upgrading machines that are running where the materials go negative (idk why)
- Just a bit more finished art would be really cool, I like the art style of the finished things
- Buff megastructures a bit

- AUTO SELL MINERALS IS BROKEN WHEN YOU GET THE UPGRADE AFTER IT
- Optimize universes
- Turning on and off shaders doesn't do anything
- There isn't an easy way to get a lot of money late game, the progression slows down
- Add auto-upgrade a bit sooner

- Fix laggy manual mining with powerful pickaxes
- Add more mining fluids
- Buff power of generated auroras 
- Add some enemy animations
- I don't really use greenhouses much

- Add a way to manually mine multiple tiles at one
- Generated wormholes are not very helpful
- Add everything speedups and maybe low level everything overclocks, its annoying at the start to spam speedups
- Make higher tier rovers much more expensive but much better
- Randomly flip tiles to reduce repetitiveness

- I don't use the subatomic system, its kinda confusing and unnecessary 
- The ship boost minigame is cool but confusing and its too weak to be important
- Add a way to search planets from the galaxy scale, for things like auroras or how big they are, ect
- Merge the shop and crafting
- Merge ships and vehicles

- Thx for reading, if you need help with the game just ask!

**Comment by Apple0726** [comment]
Hey, thanks for your comment! I'm aware of and agree with most of what you said. The game is still in a huge WIP state, and since I'm the only one working on the game, I have to prioritize important things, which means there will be a whole lot of smaller issues.

That said, I would like clarification on a couple of points:

- "Auto sell minerals is broken", how?
- "Optimize universes", what do you mean?
- "Randomly flip tiles to reduce repetitiveness" do you mean I flip the tile textures? As they are right now, the textures are seamless. I don't think they will be seamless anymore if I flipped some of them, and that would be an eye sore I think.

**Comment by ghost** [comment]
- The auto-sell works but I had to turn it on in the game menu, the in-game settings didn't work for me
- Sometimes the universe is laggy but I suppose it isn't that big of a deal
- Hmm, well I guess I just saw repetition. Maybe the textures could be embiggened and used for a larger area instead of only one block?

I do have some additional and more important bugs to report though,
- The boring machines do not work at all, especially for terraforming. They seemed to work at the start but then stopped recently, restarting doesn't fix the issue (UPDATE: I left it running overnight and then it worked again but it did stop for a while)
- The 0th stage planet-killer is buggy and doesn't work when I try to charge it
- Fighters stopped working after I conquered a few galaxies, the interface thinks I don't have any fighters even though I have plenty in the system
- Boosting in caves doesn't stop

Bonus sugguestions
- Atom manipulator should be able to make rocks (it currently only gets atoms from rocks) 
- You also can't buy rocks :(
- Maybe an achievement for buying or selling lots of items from the shop or inventory
- A research tree for buying and selling items from the inventory would be neat (cheaper prices or auto-buy materials with money)

I've been trying to create a game quite like this so when I saw Helixteus 3 I thought it was really cool, I've made a lot of plans for my game but I haven't made a finished product like this, I have some experience with coding (Javascript mainly). I can help in developing/designing if you want.

**Comment by Apple0726** [comment]
- For your boring machines issue, it might be a bug, but it also might be normal, intentional behavior. Have you built them across many planets? Since boring machines are very computationally expensive (they simulate manual mining), to avoid lag spikes, what the game does to update them is to cycle through all planets with at least one boring machine every half second, then process them. So if you have many hundreds of planets with boring machines, it might take a while before your newly terraformed boring machine planet updates and gives resources.
- The fighter issue is probably because at one point you deleted a fleet, which breaks a lot of things including your planet killer. This will be fixed in the next update

As for help developing the game, I'm interested in more ways of generating galaxies. Right now there are only two algorithms: the circular galaxy with little clusters of stars, and the spiral one. I've always wanted to add more types but for now it's pretty low priority. If that interests you, that would be cool! I can explain anything that is necessary.

**Comment by ghost** [comment]
Yep, I have a ton of boring machines. A way to fix this might be calling them extractors and instead of drilling they just produce materials from mining (doesn't even have to be what you would actually get) and have it not actually make a hole. BTW superweapons don't work except for buff and debuff. 

I'll take a look at the generation, it looks like it's in 'game.gd' lines 2075-2513. 
What other types of galaxies should there be apart from spirals and circular?

**Comment by Apple0726** [comment]
I thought about a lot of ways to simplify boring machine processing, but even if they only gave stone, the fact that tiles give more stone the deeper you are forces me to calculate all tiles individually to see what depth they are in. Or.. they could just give a constant amount of stone and nothing else, but that wouldn't be very fun..

For galaxy types, I don't really care as long as they look cool. Maybe another spiral galaxy with 4 arms?
